sqlmesh/core/engine_adapter/base.py

```diff
@@ -501,23 +576,47 @@ def _create_table_from_source_queries(
         **kwargs: t.Any,
     ) -> None:
         table = exp.to_table(table_name)
+
+        # Build a schema expression with column comments if the engine supports it
+        table_description = kwargs.pop("table_description", None)
+        column_descriptions = kwargs.pop("column_descriptions", None)
+        schema = None
+        if column_descriptions and self.COMMENT_CREATION == CommentCreation.IN_SCHEMA_DEF:
+            schema = self._build_schema_exp(
+                table, columns_to_types or self.columns(table), column_descriptions
+            )
+
         with self.transaction(condition=len(source_queries) > 1):
             for i, source_query in enumerate(source_queries):
                 with source_query as query:
                     if i == 0:
                         self._create_table(
-                            table,
+                            schema if schema else table,
                             query,
                             columns_to_types=columns_to_types,
                             exists=exists,
                             replace=replace,
+                            table_description=table_description
+                            if self.COMMENT_CREATION == CommentCreation.IN_SCHEMA_DEF
```

Comment:
This is another reason to put this closer to the source - to avoid duplication like this.